Content/Items/Forest/Armors.SlimePrince.cs

```diff
@@ -0,0 +1,273 @@
+
+				helm.targetVel += helm.targetAccel;
+
+				if (helm.targetVel.Length() > 10)
+					helm.targetVel = Vector2.Normalize(helm.targetVel) * 9.99f;
+
+				if (helm.targetAccel.Length() > 1)
+					helm.targetAccel = Vector2.Normalize(helm.targetAccel) * 0.99f;
+
+				player.velocity = helm.targetVel;
+				player.fallStart = (int)player.position.Y;
+			}
+		}
+
+		/// <summary>
+		/// Handles the prince abosrbing damage instead of damaging the player
+		/// while they are fused together
+		/// </summary>
+		/// <param name="player"></param>
+		/// <param name="info"></param>
+		/// <returns></returns>
+		private bool AbosrbSlime(Player player, Player.HurtInfo info)
+		{
+			var helm = player.armor[0].ModItem as SlimePrinceHead;
+
+			if (helm?.Minion?.Merged ?? false)
+			{
+				helm.Minion.life -= info.Damage * 4;
+				player.immuneTime = 15;
+				player.immune = true;
+
+				CombatText.NewText(player.Hitbox, Color.Gray, info.Damage);
+				SoundEngine.PlaySound(SoundID.NPCHit1, player.Center);
+
+				return true;
+			}
+
+			return false;
+		}
+	}
+
+	[AutoloadEquip(new EquipType[] { EquipType.Body, EquipType.Front })]
+	public class SlimePrinceChest : ModItem
+	{
+		public override string Texture => AssetDirectory.ForestItem + Name;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Slime Prince's Curiass");
+			Tooltip.SetDefault("5% increased summoning damage\nYou can summon an additional minion");
+
+			ArmorIDs.Body.Sets.IncludedCapeFront[Item.bodySlot] = Item.frontSlot;
+		}
+
+		public override void SetDefaults()
+		{
+			Item.width = 18;
+			Item.height = 18;
+			Item.value = 1;
+			Item.rare = ItemRarityID.Green;
+			Item.defense = 4;
+		}
+
+		public override bool IsArmorSet(Item head, Item body, Item legs)
+		{
+			return head.type == ItemType<SlimePrinceHead>() && body.type == ItemType<SlimePrinceChest>() && legs.type == ItemType<SlimePrinceLegs>();
+		}
+
+		public override void UpdateEquip(Player player)
+		{
+			player.GetDamage(DamageClass.Summon) += 0.05f;
+			player.maxMinions += 1;
+		}
+	}
+
+	[AutoloadEquip(EquipType.Legs)]
+	public class SlimePrinceLegs : ModItem
+	{
+		public override string Texture => AssetDirectory.ForestItem + Name;
+
+		public override void Load()
+		{
+			StarlightProjectile.OnHitNPCEvent += InflictExposure;
+		}
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Slime Prince's Tassets");
+			Tooltip.SetDefault("Minions inflict 5% exposure");
+		}
+
+		public override void SetDefaults()
+		{
+			Item.width = 18;
+			Item.height = 18;
+			Item.value = 1;
+			Item.rare = ItemRarityID.Green;
+			Item.defense = 3;
+		}
+
+		public override bool IsArmorSet(Item head, Item body, Item legs)
+		{
+			return head.type == ItemType<SlimePrinceHead>() && body.type == ItemType<SlimePrinceChest>() && legs.type == ItemType<SlimePrinceLegs>();
+		}
+
+		private void InflictExposure(Projectile projectile, NPC target, NPC.HitInfo hit, int damageDone)
```

Comment:
good candidate for using the new StarlightPlayer.SetHitPacketStatus(true) though the StarlightProjectile.Onhit events are new to me and not handled by it (do we really use the processing order in any meaningful capacity that it needs to be split?)

Reply:
correction on this one: it wouldn't need to be set to true in the params since this would be on the event not via a projectile's override